src/Plate/coordinate.ts

```diff
@@ -0,0 +1,131 @@
+import { CoordinateSystem } from './coordinateSystem';
+import { Coordinates, FlowDirection } from './types';
+
+export class Coordinate {
+  public row: Coordinates['row'];
+
+  public column: Coordinates['column'];
+
+  public coordinateSystem: CoordinateSystem;
+
+  constructor(
+    row: Coordinates['row'],
+    column: Coordinates['column'],
+    coordinateSystem: CoordinateSystem,
+  ) {
+    if (!coordinateSystem.rows().includes(row)) {
+      throw new Error(
+        `Expected a row with value of ${coordinateSystem
+          .rows()
+          .join(',')}, got ${row}.`,
+      );
+    }
+    this.row = row;
+
+    if (!coordinateSystem.columns().includes(column)) {
+      throw new Error(
+        `Expected a column with value of ${coordinateSystem
+          .columns()
+          .join(',')}, got ${column}.`,
+      );
+    }
+    this.column = column;
+
+    this.coordinateSystem = coordinateSystem;
+  }
+
+  static fromString(
+    coordinateString: string,
+    coordinateSystem: CoordinateSystem,
+  ): Coordinate {
+    const rows = coordinateSystem.rows();
+    const rowsOptions = rows.join('|');
+
+    const columns = [
+      ...coordinateSystem.columns(),
+      ...coordinateSystem.paddedColumns(),
+    ];
+    const columnsOptions = columns.join('|');
+
+    const matches = coordinateString.match(
+      new RegExp(`^(${rowsOptions})(${columnsOptions})`),
+    );
+
+    if (matches === null || matches?.length === 0) {
+      const firstValidExample = rows[0].toString() + columns[0].toString();
```

Comment:
If `rows` or `columns` is empty an error would be thrown here:

![image](https://user-images.githubusercontent.com/4804412/161541351-f8ffc60e-08b0-4f71-bfc2-d9511f39a41c.png)

So there should be a check for the length of the array. You might also want to use `lodash` -> `first()` for this.

This would be highlighted by Typescript if we would enable  `noUncheckedIndexedAccess` in `tsconfig.json`. We should discuss that at some point.

Edit: `noUncheckedIndexed` should already be active because of https://github.com/mll-lab/tsconfig/blob/master/tsconfig.json . I wonder why TS does not complain here